test/SILOptimizer/existential_specializer_soletype.sil

```diff
@@ -0,0 +1,138 @@
+
+// This file tests existential specializer transformation followed by concrete type propagation  and generic specialization leading to a devirtualization of a witness method call.
+
+sil_stage canonical
+
+import Builtin
+import Swift
+import SwiftShims
+
+protocol RP {
+  func getThres() -> Int32
+}
+
+class RC : RP {
+  @inline(never) func getThres() -> Int32
+}
+
+sil hidden [noinline] @$s6simple2RCC8getThress5Int32VyF : $@convention(method) (@guaranteed RC) -> Int32 {
+bb0(%0 : $RC):
+  %1 = integer_literal $Builtin.Int32, 10
+  %2 = struct $Int32 (%1 : $Builtin.Int32)
+  return %2 : $Int32
+} 
+
+// CHECK-LABEL: sil hidden [noinline] @$s6simple12find_wrapperSbyF : $@convention(thin) () -> Bool {
+// CHECK: bb0:
+// CHECK:   alloc_ref
+// CHECK:   integer_literal
+// CHECK:   struct
+// CHECK:   alloc_stack
+// CHECK:   store
+// CHECK:   strong_retain
+// CHECK:   function_ref @$s6simple4find5count4Obj1Sbs5Int32V_AA2RP_ptFTf4ne_n4main2RCC_Tg5 : $@convention(thin) (Int32, @guaranteed RC) -> Bool
+// CHECK:   load
+// CHECK:   apply
+// CHECK:   destroy_addr
+// CHECK:   dealloc_stack
+// CHECK:   strong_release
+// CHECK:   return
+// CHECK-LABEL: } 
+sil hidden [noinline] @$s6simple12find_wrapperSbyF : $@convention(thin) () -> Bool {
+bb0:
+  %0 = alloc_ref $RC                              
+  %3 = integer_literal $Builtin.Int32, 0          
+  %4 = struct $Int32 (%3 : $Builtin.Int32)        
+  %5 = alloc_stack $RP                            
+  %6 = init_existential_addr %5 : $*RP, $RC       
+  store %0 to %6 : $*RC                           
+  %8 = function_ref @$s6simple4find5count4Obj1Sbs5Int32V_AA2RP_ptF : $@convention(thin) (Int32, @in_guaranteed RP) -> Bool 
+  strong_retain %0 : $RC                          
+  %10 = apply %8(%4, %5) : $@convention(thin) (Int32, @in_guaranteed RP) -> Bool 
+  destroy_addr %5 : $*RP                          
+  dealloc_stack %5 : $*RP                         
+  strong_release %0 : $RC                         
+  return %10 : $Bool                              
+} 
+
+sil private [transparent] [thunk] @$s6simple2RCCAA2RPA2aDP8getThress5Int32VyFTW : $@convention(witness_method: RP) (@in_guaranteed RC) -> Int32 {
+bb0(%0 : $*RC):
+  %1 = load %0 : $*RC 
+  %2 = class_method %1 : $RC, #RC.getThres!1 : (RC) -> () -> Int32, $@convention(method) (@guaranteed RC) -> Int32
+  %3 = apply %2(%1) : $@convention(method) (@guaranteed RC) -> Int32 
+  return %3 : $Int32                              
+} 
+
+
+// CHECK-LABEL: sil shared [noinline] @$s6simple4find5count4Obj1Sbs5Int32V_AA2RP_ptFTf4ne_n4main2RCC_Tg5 : $@convention(thin) (Int32, @guaranteed RC) -> Bool {
+// CHECK: bb0
+// CHECK:   alloc_stack
+// CHECK:   store
+// CHECK:   alloc_stack
+// CHECK:   init_existential_addr
+// CHECK:   copy_addr
+// CHECK:   open_existential_addr
+// CHECK:   unchecked_addr_cast
+// CHECK:   load
+// CHECK:   function_ref @$s6simple2RCC8getThress5Int32VyF : $@convention(method) (@guaranteed RC) -> Int32 
```

Comment:
this has been devirtualized after existential specialization transformation is applied.

Reply:
Done